### PR TITLE
Hotfix - iOS Sharing

### PR DESCRIPTION
**Pull request by johnhaitas**
* iOS sharing is currently broken
  - Can be observed in `0.5.1` by attempting the *Copy* action - no text will be copied.

### CODE REVIEW

File: packages/share/ios/Classes/SharePlugin.m [hunk 1]
```diff
@@ -16,8 +16,12 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   [shareChannel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
     if ([@"share" isEqualToString:call.method]) {
       NSDictionary *arguments = [call arguments];
+      NSString *shareText;
+      if ([arguments[@"text"] isKindOfClass:[NSString class]] == YES) {
```
Comment (mravn-google) [inline packages/share/ios/Classes/SharePlugin.m]: I suppose we can simplify the condition to
```objective-c
if ([arguments[@"text"] isKindOfClass:[NSString class]]) { ... }
```
However, since the Dart/Java/ObjC sides of a plugin are co-developed, our plugins don't explicitly validate the arguments received. Instead we assume everything is fine on the receiver side, like so:
```objective-c
NSString *shareText = call.arguments[@"text"];
[self share:shareText withController:...];
```
In particular, the type check and the checks for a null or empty string are already done [on the Dart side](https://github.com/flutter/plugins/blob/master/packages/share/lib/share.dart#L31-L32), so there is little point in repeating that here.

Comment (johnhaitas) [reply packages/share/ios/Classes/SharePlugin.m]: @mravn-google I have a habit of being extremely cautious with type-safety. This patch could certainly be written in a more succinct way. Let me know if you prefer it with less lines of code.

Comment (mravn-google) [reply packages/share/ios/Classes/SharePlugin.m]: @johnhaitas I would prefer it written more succinctly to match other plugins in this repo, if you don't mind.

Comment (johnhaitas) [reply packages/share/ios/Classes/SharePlugin.m]: @mravn-google done with https://github.com/flutter/plugins/pull/602/commits/bf0af0d72b2a792895941d1db2881cc5f44fa54c

File: packages/share/ios/Classes/SharePlugin.m [hunk 2]
```diff
@@ -16,8 +16,9 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   [shareChannel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
     if ([@"share" isEqualToString:call.method]) {
       NSDictionary *arguments = [call arguments];
+      NSString *shareText = arguments[@"text"];
 
-      if ([arguments[@"text"] length] == 0) {
+      if (shareText.length == 0) {
```
Comment (mravn-google) [inline packages/share/ios/Classes/SharePlugin.m]: You might consider removing this check too. It is already checked on the Dart side. Or we can leave that for another PR. You call.

Comment (johnhaitas) [reply packages/share/ios/Classes/SharePlugin.m]: @mravn-google i deliberately left that logic as is - just replaced `arguments[@"text"]` with the unpacked `shareText` variable. That line is the main reason I opted for the `if ([arguments[@"text"] isKindOfClass:[NSString class]] == YES) {` type check, because it could throw an exception if `shareText` is not `nil` or an `NSString` (for example: `[NSNull null]`).

**Comment by johnhaitas** [comment]
@mravn-google what is the next step? Can someone merge this in?

File: packages/share/ios/Classes/SharePlugin.m [hunk 3]
```diff
@@ -16,8 +16,9 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   [shareChannel setMethodCallHandler:^(FlutterMethodCall *call, FlutterResult result) {
     if ([@"share" isEqualToString:call.method]) {
       NSDictionary *arguments = [call arguments];
+      NSString *shareText = arguments[@"text"];
```
Comment (xster) [inline packages/share/ios/Classes/SharePlugin.m]: I might be misunderstanding the nature of this PR. UIActivityViewController's initWithActivityItems expects an NSArray. 

Does the example code not work? Seems to work ok in my testing.

**Comment by mravn-google** [comment]
I'll publish the new version asap.

File: packages/share/ios/Classes/SharePlugin.m (hunk 3 above)
Comment (johnhaitas) [reply packages/share/ios/Classes/SharePlugin.m]: @xster it's added to an array inside the `+share:withController:atSource:` class method
https://github.com/flutter/plugins/pull/602/files#diff-95f220dbd86cfc79628c3b434c755122R52

This was existing functionality and has not been modified

Comment (xster) [reply packages/share/ios/Classes/SharePlugin.m]: Ah, you're right. Thanks
